docs/training/apis-howto/api-core-mnist-ug.rst

```diff
@@ -0,0 +1,573 @@
+#####################
+ Core API User Guide
+#####################
+
+This guide will help you get up and running with the Core API.
+
++------------------------------------------------------------------+
+| Visit the API reference                                          |
++==================================================================+
+| :doc:`/reference/reference-training/training/api-core-reference` |
++------------------------------------------------------------------+
+
+In this user guide, we'll show you how to adapt model training code to use the Core API. As an
+example, we'll be working with the PyTorch MNIST model.
+
+************
+ Objectives
+************
+
+The step-by-step instructions walk you through modifying a script for the purpose of performing the
+following functions:
+
+-  Report metrics
+-  Report checkpoints
+-  Perform a hyperparameter search
+-  Perform distributed training
+
+After completing the steps in this user guide, you will be able to do the following:
+
+-  Understand the minimum requirements for running an experiment
+-  Modify a script and an experiment configuration file
+-  Understand how to convert model code
+-  Use the Core API to train a model
+
+***************
+ Prerequisites
+***************
+
+**Required**
+
+-  A Determined cluster
+
+**Recommended**
+
+-  :doc:`Quickstart for Model Developers </quickstart-mdldev>`
+
+*****************************************************
+ Step 1: Get the Tutorial Files & Run the Experiment
+*****************************************************
+
+To run an experiment, you need, at minimum, a script and an experiment configuration (YAML) file.
+
+Create a new directory.
+
+Get the files from the :download:`core_api_pytorch_mnist.tgz </examples/core_api_pytorch_mnist.tgz>`
+download or from the `Github repository
+<https://github.com/determined-ai/determined/tree/master/examples/tutorials/core_api_pytorch_mnist>`_.
+
+.. note::
+
+   Throughout this user guide, we’ll show you how to make modifications to your script. After each
+   step, we’ll re-run our experiment using the appropriate script and its accompanying experiment
+   configuration (YAML) file.
+
+In this initial step, we’ll run our experiment using the ``model_def.py`` script and its
+accompanying ``const.yaml`` experiment configuration file.
+
+CD into the directory and run this command:
+
+.. code::
+
+   det e create const.yaml . -f
+
+Open the Determined WebUI by navigating to ``http://localhost:8080/``. Accept the default determined
+username, leave the password empty, and click **Sign In**.
+
+In the WebUI, select your experiment, and then navigate to the **Logs** tab.
+
+************************
+ Step 2: Report Metrics
+************************
+
+To report training and validation metrics to the Determined master, we’ll add a few lines of code to
+our script. More specifically, we'll create a ``core.Context`` object to allow interaction with the
```

Comment:
All of these references to objects need to be links to reference docs.  That's what is so amazing about sphinx-generated docs, is as a reader, I can read through a tutorial and as soon as I say, "hey, what's that object?" I can click on a link and it takes me right to the reference docs that explain exactly what that document is.

Looking at my own doc, I was not great about linking all of the methods I referenced, but IIRC I wrote that document in a serious hurry.  It should have been better cross-linked.

Reply:
to be sure, the content would become

```
To report training and validation metrics to the Determined master, we’ll add a few lines of code to
our script. More specifically, we'll create a :class:`~determined.core.Context` object to allow interaction with the
```

Reply:
@rb-determined-ai I applied the text role to the instance discussed here and did not find any additional instances in the doc. maybe i missed something.